invenio_app_rdm/utils/files.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio App RDM is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Files utilities."""
+
+from __future__ import absolute_import, print_function
+
+from datetime import datetime
+
+import sqlalchemy as sa
+from flask import current_app
+from flask_mail import Message
+from invenio_files_rest.models import FileInstance
+from invenio_rdm_records.records.models import RDMDraftMetadata, RDMRecordMetadata
+
+
+def checksum_verification_files_query():
+    """Return a FileInstance query taking into account file URI prefixes."""
+    files = FileInstance.query
+    uri_prefixes = current_app.config.get(
+        "FILES_REST_CHECKSUM_VERIFICATION_URI_PREFIXES"
+    )
+    if uri_prefixes:
+        files = files.filter(
+            sa.or_(*[FileInstance.uri.startswith(p) for p in uri_prefixes])
+        )
+    return files
+
+
+def render_email_from_context(template, context):
+    """Renders a template using a provided context."""
+    template = current_app.jinja_env.get_or_select_template(template)
+    return template.render(context)
+
+
+def generate_integrity_report_entries(files):
+    """Generates the report entries."""
+    entries = []
+    for file in files:
+        entry = _generate_report_entry(file)
+        entries.append(entry)
+    return entries
+
+
+def _generate_report_entry(file):
+    """Generates a single report for a file."""
+    entry = {"file": file}
+    for o in file.objects:
+        entry["filename"] = o.key
+
+        # Find records/deposits for the files
+        deposit = get_deposit_from_bucket(o.bucket_id)
```

Comment:
I am maybe mistaken, but in RDM we use more the term `draft` than `deposit` (old). Shall we use the same name around?
```suggestion
        deposit = get_draft_from_bucket(o.bucket_id)
```